src/js/components/FormPasswordInput/FormPasswordInput.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import Button from '../Button/Button';
+
+type Props = {
+  className?: string,
+};
+
+type State = {
+  hidePassword: boolean,
+};
+
+class FormPasswordInput extends React.Component<Props, State> {
+  state = {
+    hidePassword: true,
+  };
+
+  togglePassword() {
+    this.setState({
+      hidePassword: !this.state.hidePassword,
+    });
+  }
+
+  render() {
+    const { className, ...other } = this.props;
+    const { hidePassword } = this.state;
+    const classes = classNames(className, 'form-password-input', 'input-group');
+
+    const inputType = hidePassword ? 'password' : 'text';
+    const buttonText = hidePassword ? 'Show' : 'Hide';
+    const icon = hidePassword ? 'preview' : 'previewHide';
+
+    return (
+      <div className={classes}>
+        <input type={inputType} className="form-control" {...other} />
```

Comment:
This doesn't work as expected. In `FormStories.js` where you are using this element you are passing `name="password" type="password"` as props. Because you are spreading props with `{...other}` these are overriding the `type` attribute which the state controls so the type is always set to password. I guess you could filter props to make sure that default HTML attributes don't make there way through or just remove the spread. Maybe there is a solution with using types but I couldn't find anything on with a quick Google.

Reply:
I would put `{...other}` first so that you can't override `type` and `className`.

Reply:
We don't want to be able to override type because this is genuinely a password field and seems to default the object of encapsulated components. We can already override `className`.

Reply:
That's why you need to put `{...other}` first.  Having it last in the list of props means, if a user passes `type` or `className` as a prop it will override your defaults.

Currently the `className` override exposed by this component is being applied to the `div` on the line above, not sure whether that's what we want or not though...

Reply:
My feeling is maybe it should be the same as the `SearchInput` component in Ray.

Reply:
Yes, I went back and forth on this, but decided it should be for the wrapper because it is for the whole component.

Sorry I misunderstood so will fix up now.